airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -123,38 +150,33 @@ def _configure_logger_level(self, logger: logging.Logger):
-        full_config["check"] = self._source_config["check"]
-        streams = [self._factory.create_component(stream_config, {}, False)() for stream_config in self._stream_configs()]
-        if len(streams) > 0:
-            full_config["streams"] = streams
-        declarative_source_schema = ConcreteDeclarativeSource.json_schema()
-
-        try:
-            validate(full_config, declarative_source_schema)
-        except ValidationError as e:
-            raise ValidationError("Validation against auto-generated schema failed") from e
-
-        # Validates the connector manifest against the low-code component json schema
-        manifest = self._source_config
-        if "type" not in manifest:
-            manifest["type"] = "DeclarativeSource"
-        manifest_transformer = ManifestComponentTransformer()
-        propagated_manifest = manifest_transformer.propagate_types_and_options("", manifest, {})
-
-        try:
-            raw_component_schema = pkgutil.get_data("airbyte_cdk", "sources/declarative/declarative_component_schema.yaml")
-            declarative_component_schema = yaml.load(raw_component_schema, Loader=yaml.SafeLoader)
-        except FileNotFoundError as e:
-            raise FileNotFoundError(f"Failed to read manifest component json schema required for validation: {e}")
-
-        try:
-            validate(propagated_manifest, declarative_component_schema)
-        except ValidationError as e:
-            raise ValidationError("Validation against json schema defined in declarative_component_schema.yaml schema failed") from e
+        if self.construct_using_pydantic_models:
+            # Validates the connector manifest against the low-code component schema defined in declarative_component_schema.yaml
+            try:
+                raw_component_schema = pkgutil.get_data("airbyte_cdk", "sources/declarative/declarative_component_schema.yaml")
+                declarative_component_schema = yaml.load(raw_component_schema, Loader=yaml.SafeLoader)
+            except FileNotFoundError as e:
+                raise FileNotFoundError(f"Failed to read manifest component json schema required for validation: {e}")
+
+            try:
+                validate(self._source_config, declarative_component_schema)
+            except ValidationError as e:
+                raise ValidationError("Validation against json schema defined in low_code_component_schema.yaml schema failed") from e
+        else:
+            # Validates the connector manifest against the one schema auto-generated from the component dataclasses
+            full_config = {}
+            if "version" in self._source_config:
+                full_config["version"] = self._source_config["version"]
+            full_config["check"] = self._source_config["check"]
+            streams = [self._factory.create_component(stream_config, {}, False)() for stream_config in self._stream_configs()]
+            if len(streams) > 0:
```

Comment:
question: what should we do / how should we fail if there are no streams?

Reply:
Good question. In the context of this code, we're just appending the constructed declarative streams back to the object for validation. We'll end up deleting this code in the long run though so I don't think we need to change anything here.

That being said, we should be able to update the json schema to enforce a minimum of 1 element and avoid having a specific conditional check in our code.

Reply:
Annoyingly, it looks like there's a bug in pydantic that causes an error when we try to use field constraints on generics. 
```
E   ValueError: On field "streams" the following field constraints are set but not enforced: min_items. 
E   For more details see https://pydantic-docs.helpmanual.io/usage/schema/#unenforced-field-constraints
```

There's a bug filed here w/ the exact use case, in the meantime I'll have to get rid of the field constraint because the way the model generator works, it will always emit classes in the format that causes the error until it is fixed:
https://github.com/pydantic/pydantic/issues/3358